src/insexpand.c

```diff
@@ -2579,6 +2580,7 @@ ins_compl_prep(int c)
 {
     int		retval = FALSE;
     int		prev_mode = ctrl_x_mode;
+    int		handle_enter = c == CAR && compl_selected_item == -1;
```

Comment:
Do we need to consider `c == NL` and/or `c == K_KENTER`?
```suggestion
    int		handle_enter = FALSE;
    if ((c == CAR || c == NL || c == K_KENTER) && compl_selected_item == -1)
        handle_enter = TRUE;
```